care/users/api/viewsets/users.py

```diff
@@ -387,3 +400,40 @@ def check_availability(self, request, username):
         if User.check_username_exists(username):
             return Response(status=status.HTTP_409_CONFLICT)
         return Response(status=status.HTTP_200_OK)
+
+    def has_profile_image_write_permission(self, user, request):
+        allowed_user_types = [
+            User.TYPE_VALUE_MAP["WardAdmin"],
+            User.TYPE_VALUE_MAP["LocalBodyAdmin"],
+            User.TYPE_VALUE_MAP["DistrictAdmin"],
+            User.TYPE_VALUE_MAP["StateAdmin"],
+        ]
+        return (
+            user.is_superuser
+            or (user.id == request.user.id)
+            or (
+                user.user_type in allowed_user_types
+                and self.facility.has_object_write_permission(request)
```

Comment:
how can you access self.facility? I think it should be user.home_facility

you also need to check if the request.user's role is higher than the user